### PR TITLE
[Core] ModelPart Check for nodes only if Variable was not added yet

### PR DESCRIPTION
**Pull request by philbucher**
This PR adds an `if` to only throw the error if the Variable does not exist yet. 

The VariableList checks internally if the Variable was added yet so the function in the ModelPart can be called safely

Sorry for the whitespace changes

@jcotela as discussed

### CODE REVIEW

File: kratos/includes/model_part.h [hunk 1]
```diff
@@ -469,8 +469,11 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     template<class TDataType>
     void AddNodalSolutionStepVariable(Variable<TDataType> const& ThisVariable)
     {
-        KRATOS_ERROR_IF((this->GetRootModelPart()).Nodes().size() != 0) << "attempting to add the variable: " << ThisVariable.Name() << " to the model part with name: " << this->Name() << " which is not empty" << std::endl;
-        mpVariablesList->Add(ThisVariable);
+        if (!mpVariablesList->Has(ThisVariable))
```
Comment (RiccardoRossi) [inline kratos/includes/model_part.h:472]: oh i see.

however i am not sure if we should throw an error at all here. maybe we should create the mpVariableList if it does not exist. @KratosMultiphysics/technical-committee opinions?

**Review by RiccardoRossi — COMMENTED** [review]
hi @philbucher i am suggesting to test the error message. this is useful if we make changes in the future

take a look at 

https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/tests/test_model.py#L29

for an example

File: kratos/includes/model_part.h (hunk 1 above)
Comment (philbucher) [reply kratos/includes/model_part.h:472]: @RiccardoRossi  we had quite some problems before this error was added (done by you btw), see #492

**Comment by philbucher** [comment]
@RiccardoRossi added the test

File: kratos/includes/model_part.h (hunk 1 above)
Comment (RiccardoRossi) [reply kratos/includes/model_part.h:472]: jeje, i have short memory...thx for the pointer

**Review by RiccardoRossi — COMMENTED** [review]
ok, this looks good to me. @KratosMultiphysics/technical-committee if you have no comments i'll approve tomorrow

**Comment by philbucher** [comment]
Ping @RiccardoRossi

**Review by RiccardoRossi — APPROVED** [review]
@roigcarlo can we still merge this to the release?

**Comment by philbucher** [comment]
Thx

**Pull request merged**